### ISSUE TITLE
Examples outdated

### ISSUE THREAD

**Issue by m-mohr** [Open-EO/openeo.org#16]
The examples are outdated: https://openeo.org/documentation/1.0/developers/examples/
Doesn't even use the new processes, but simply shows process graphs in the old pre 1.0.0-rc.2 state.

**Comment by m-mohr** [comment]
People sometimes "complain" that there should be more examples, smaller examples derived from real world examples you can go through, to understand how the processes work. Like how to work with reducers, merging, generally how to approach issues. Many have no experience with data cubes and related processes. We could have a guide for all languages at once, showing the code snippets for all languages in tabs or so and then go through various things.

**Comment by jonathom** [comment]
I want to open a discussion about this idea of having more organized examples, so if you have any comments and ideas on what should be explained there, please post! Please also mention people you feel could add something. @m-mohr @christophfriedrich @edzer 

**Outcome:** An openEO cookbook users can refer to to get a first idea on how to solve their problems with the openEO API in the three languages Python, R and JavaScript (as described above). It should not replace the [getting started guides](https://openeo.org/documentation/1.0/javascript/) nor go as much into API details as API documentation (e.g. [python](https://open-eo.github.io/openeo-python-client/index.html)) or go through multiple complete [use cases](https://github.com/Open-EO/openeo-usecases), but should describe how to implement simple use cases in a very pragmatic way. 

**Content**:
  * Input Data: How to upload and use locally stored data?
  * Output: Formats etc., How to create a PNG?
  * Datacube core functions: haven't thought of specific uses yet, e.g. How to apply a moving average process?
  * Bandmath: How to do bandmath, how to select bands, how to make use of predefined processes like `normalized_difference`?
  * Combine Sensors: How to integrate different data sources?
  * Masking: How to mask, How to construct a mask?
  * Classification: How to make a classification?
  * Change Detection: How to do a simple change detection?
  * Zonal statistics: How to create and download a vector cube?
  * UDFs: How to make use of UDFs, how and when to write a UDF?
  * ...

**Comment by m-mohr** [comment]
See #28 for a Code Switcher example.

**Comment by jonathom** [comment]
very nice, example works like a charm @m-mohr.

**Comment by jonathom** [comment]
just a note to myself: should the cookbook contain FAQs with links to where these questions are explained (might make navigation, topics easier to find) and/or tasks at the end of a chapter to explore the used processes (a la what happens when you do this and that).

**Comment by clausmichele** [comment]
> 
> 
> just a note to myself: should the cookbook contain FAQs with links to where these questions are explained (might make navigation, topics easier to find) and/or tasks at the end of a chapter to explore the used processes (a la what happens when you do this and that).

Maybe a nice addition could be, after the correct example, some common mistakes to avoid. Like forgetting to insert the dimension in a reduce_spatial process. Since the logs from the back-ends are not always so informative, it's important to know what could go wrong.

**Comment by jonathom** [comment]
@clausmichele yes, great idea! it is already in the making - if you have anything else that strikes you as a common mistake, let me know!
